sqlglot/optimizer/eliminate_subqueries.py

```diff
@@ -144,8 +144,9 @@ def _eliminate_derived_table(scope, existing_ctes, taken):
     name, cte = _new_cte(scope, existing_ctes, taken)
 
     table = exp.alias_(exp.table_(name), alias=parent.alias or name)
-    parent.replace(table)
+    table.set("joins", parent.args.get("joins"))
 
+    parent.replace(table)
     return cte
```

Comment:
This was changed in order to correctly handle the following query:

```sql
SELECT * FROM ((SELECT c FROM t1) JOIN t2)
```

Notice that we're factoring out `SELECT c FROM t1` into a CTE and then calling `replace(table)` on it. This has the unintended side effect of getting rid of the `joins` that were initially attached on the subquery.

The problem is shown below, where the joins haven't been reattached to the table:

```python
>>> from sqlglot.optimizer import optimize
>>> optimize("SELECT * FROM ((SELECT c FROM t1) JOIN t2)").sql()
'WITH "_q_0" AS (SELECT "t1"."c" AS "c" FROM "t1" AS "t1") SELECT * FROM ("_q_0" AS "_q_0")'
```